src/function.ts

```diff
@@ -1044,7 +1051,120 @@ export async function serialize(
+                          );
+                        }
+                        return eraseBindAndRegister(entry1.right);
+                      } else if (commandFlag === RegisterRefCommand) {
+                        // Register Ref - remove
+                        // ("REGISTER_REF", ref[Symbol.AST]=ast)
+                        // => undefined
+                        // TODO support returning no node.
+                        return ts.factory.createIdentifier("undefined");
+                      } else if (commandFlag === BindCommand) {
+                        /**
+                         * Bind -
+                         * ("BIND",
+                         *    stash={ args: args, this: this, func: func },
+                         *    stash={ f: stash.func.bind(stash.self, ...stash.args), ...stash },
+                         *    typeof stash.f === "function" && (
+                         *        stash.f[Symbol.for("functionless:BoundThis")] = stash.self,
+                         *        stash.f[Symbol.for("functionless:BoundArgs")] = stash.args,
+                         *        stash.f[Symbol.for("functionless:TargetFunction")] = stash.func
+                         *    ),
+                         *    stash.func
+                         * )
+                         * => func.bind(this, ...args)
+                         */
+                        if (
+                          !entry1 ||
+                          !ts.isBinaryExpression(entry1) ||
+                          entry1.operatorToken.kind !==
+                            SyntaxKind.EqualsToken ||
+                          !ts.isObjectLiteralExpression(entry1.right)
+                        ) {
+                          throw new SynthError(
+                            ErrorCodes.Unexpected_Error,
+                            "Compilation Error: found an invalid register command. Check the versions of AST-Reflection and Functionless."
+                          );
+                        }
+                        const {
+                          args,
+                          this: _this,
+                          func,
+                        } = Object.fromEntries(
+                          entry1.right.properties
+                            .filter(ts.isPropertyAssignment)
+                            .map((p) => [
+                              p.name && ts.isIdentifier(p.name)
+                                ? p.name.text
+                                : "UNKNOWN",
+                              p.initializer,
+                            ])
+                        );
+                        if (!args || !_this || !func) {
+                          throw new SynthError(
+                            ErrorCodes.Unexpected_Error,
+                            "Compilation Error: found an invalid register command. Check the versions of AST-Reflection and Functionless."
+                          );
```

Comment:
Why not just make it a no-op?

Reply:
My thought was: The user wouldn't know it is an issue until runtime. If we cannot re-construct the command then we should fail.

Reply:
But how do you know it's not just someone else using similar looking syntax? The compiler should not produce an invalid contract right?

Reply:
We check for our flag on the left first. The chance someone has stash_salt=bind is very low. 

That leaves all parsing errors as a bug or mis-matched package version ( with breaking changes)